pkg/cloudprovider/aws/validation.go

```diff
@@ -0,0 +1,45 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+)
+
+// Validate cloud provider specific components of the cluster spec
+func (c *Capacity) Validate(ctx context.Context) error {
```

Comment:
I'm not sure what to do with this, it seems weird to have it be a method on Capacity... but not sure where else to put it.  Maybe it deserves to be its own Type "Validator", but it seems that `Capacity` is the only view into a cloudprovider right now so, not sure if there's an intended design for other types or not.

Reply:
I think ultimately a `CloudProvider.ValidatorFor()` method might be warranted, but seems like we could wait on that?

Reply:
Hard agree.

Reply:
Okay, we'll leave as is for now and potentially refactor later.